src/Relude/Extra/Enum.hs

```diff
@@ -67,10 +67,8 @@ Singleton :| []
 
 @since 0.7.0.0
 -}
-universeNonEmpty :: forall a . (Bounded a, Enum a, Eq a) => NonEmpty a
-universeNonEmpty
-    | minBound @a == maxBound = minBound :| []
-    | otherwise = minBound :| [succ minBound .. maxBound]
+universeNonEmpty :: forall a . (Bounded a, Enum a) => NonEmpty a
+universeNonEmpty = minBound :| drop 1 universe
```

Comment:
I'd be happy to rewrite in a self-contained way (e.g. explicit inline of `universe`) if that's preferred.

Reply:
I think that this is pretty neat! Thanks 👍🏼